src/pushsource/_impl/backend/koji_source.py

```diff
@@ -509,6 +521,118 @@ def _push_items_from_container_build(self, nvr, meta):
+            raise ValueError(message)
+
+        build_id = meta["id"]
+
+        archives = self._get_archives(build_id)
+
+        # Supported types of VMI archives
+        ami_types = ["raw", "raw-xz"]
+        azure_types = ["vhd", "vhd-compressed"]
+        cloud_types = {}
+
+        for k in ami_types:
+            cloud_types.setdefault(k, AmiPushItem)
+
+        for k in azure_types:
+            cloud_types.setdefault(k, VHDPushItem)
+
+        # Collect all image archives which are not containers (tar)
+        # FIXME: Need to verify whether it's better to limit only to
+        # known extensions (like VHD, AMI) or not
+        vmi_archives = [
+            elem
+            for elem in archives
+            if elem.get("btype") == "image" and elem.get("type_name") != "tar"
+        ]
+
+        out = []
+
+        # Generate the respective PushItem type for each archive
+        for archive in vmi_archives:
+            path = self._pathinfo.typedir(meta, archive["btype"])
+            item_src = os.path.join(path, archive["filename"])
+
+            extra = archive.get("extra") or {}
+            image = extra.get("image") or {}
+            arch = image.get("arch")
+
+            klass = cloud_types.get(archive["type_name"]) or VMIPushItem
+
+            rel_klass = AmiRelease if klass == AmiPushItem else VMIRelease
+
+            splitted_name = meta["name"].split("-")
+            if len(splitted_name) > 1:
+                product = "-".join(splitted_name[:-1])
+            else:
+                product = splitted_name[0]
+
+            release = rel_klass(
+                arch=arch,
+                date=meta.get("completion_time", "").split(" ")[0],
+                product=product.upper(),
+                version=meta["version"],
+                respin=int(meta["release"]),
+            )
+
+            out.append(
+                # FIXME: Is it better to ask osbuild to export the missing info in metadata?
+                #
+                # Remaining data to be gattered from an external service:
+                #
+                # AmiPushItem:
+                #   - type
+                #   - region
+                #   - virtualization
+                #   - volume
+                #   - root_device
+                #   - sriov_net_support
+                #   - ena_support
+                #   - billing_codes
+                #
+                # VHDPushItem:
+                #   - generation
+                #   - sku_id
+                #   - support_legacy
+                #   - legacy_sku_id
+                #   - disk_version
+                #   - recommended_sizes
+                #   - sas_uri
+                klass(
+                    name=archive["filename"],
+                    description="",
+                    dest=self._dest,
+                    src=item_src,
+                    build=nvr,
+                    build_info=KojiBuildInfo(
+                        name=meta["name"],
```

Comment:
@Tojaj @lslebodn @jajreidy this is the updated list of missing metadata.

For `Azure` the at least the following metadata must be present before publishing:

- disk_version: should come from somewhere: OSBS or StArMap.
- sas_uri (will come from upload)

The other ones have the following defaults:
- generation: Defaults to `V2`
- sku_id: Defaults to `{plan_name}`
- support_legacy: Defaults to `False`
- legacy_sku_id: Defaults to `{plan_name}-gen1`
- recommended_sizes: Defaults to `[]`

@jajreidy do you have the list of required metadata for AWS as well? Maybe we can update the doc with all this information, WDYT?